src/miner.erl

```diff
@@ -600,9 +445,234 @@ terminate(Reason, _State) ->
     lager:info("stopping: ~p", [Reason]),
     ok.
 
-%% ==================================================================
-%% Internal functions
-%% =================================================================
+%% ============================================================================
+%% Private
+%% ============================================================================
+
+-spec try_create_block(
+    metadata(),
+    blockchain_txn:txns(),
+    non_neg_integer(),
+    blockchain:blockchain(),
+    swarm_keys()
+) ->
+    create_block_result().
+try_create_block(Metadata, Txns, HBBFTRound, Chain, SwarmKeys) ->
+    %% This can actually be a stale message, in which case we'd produce a block
+    %% with a garbage timestamp. This is not actually that big of a deal, since
+    %% it won't be accepted, but we can short circuit some effort by checking
+    %% for a stale hash.
+    Chain = blockchain_worker:blockchain(),
```

Comment:
why are we refetching the chain?

Reply:
Great catch! Artifact of moving things around...

Reply:
Fixed